server/container/util-java8/src/test/java/org/apache/james/util/streams/SwarmGenericContainer.java

```diff
@@ -21,40 +21,92 @@
 
 import java.util.List;
 
+import org.junit.Assume;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.images.builder.ImageFromDockerfile;
 
 import com.google.common.base.Strings;
+import org.testcontainers.shaded.com.github.dockerjava.api.command.InspectContainerResponse;
 
-public class SwarmGenericContainer extends GenericContainer<SwarmGenericContainer> {
-
+public class SwarmGenericContainer implements TestRule {
     private static final Logger LOGGER = LoggerFactory.getLogger(SwarmGenericContainer.class);
     private static final String DOCKER_CONTAINER = "DOCKER_CONTAINER";
+    private static final String NO_DOCKER_ENVIRONMENT = "Could not find a valid Docker environment.";
+    private static final String SKIPPING_TEST_CAUTION = "Skipping all docker tests as no Docker environment was found";
+
+    private GenericContainer container;
 
     public SwarmGenericContainer(String dockerImageName) {
-        super(dockerImageName);
+        try {
+            this.container = new GenericContainer(dockerImageName);
+        } catch (IllegalStateException e) {
+            LOGGER.error("Cannot initial a docker container because: " + e.getMessage());
```

Comment:
Log the full exception, not just the message. People would need to know what's happening.